### PR TITLE
upgraded default react-native version

### PR DESCRIPTION
**Pull request by lvlrSajjad**
upgraded default react-native version to 0.57.7

updated react-navigation to 3.0.0

also updated ignite-dev-screens,redux,redux-persist and some other libs to the last version

after changes i got it running in android and it was ok

removed allowFontScaling from `boilerplate/App/Config/index.js` cause it causes error in new react-native version

### CODE REVIEW

**Comment by lvlrSajjad** [comment]
in semaphore log it seems we getting errors like these
```
Yarn supports the following semver range: "^4.8.0 || ^5.7.0 || ^6.2.2 || >=8.0.0"\nerror react-native@0.57.5: The engine "node" is incompatible with this module. Expected version ">=8.3".\nerror Found incompatible module\n{ Error: Command failed: yarn add react-native@0.57.5
```
```
Command `yarn add react-native@0.57.5 --exact` failed
```

i think the semaphor's yarn version is in incompatible with react-native@0.57.5.

in my system i can run this test at least the first one successfully passes .

image

**Comment by lvlrSajjad** [comment]
Just ignite-dev-screens may not work with this version of react-navigation so if you merge this , i will upgrade the plugin too

**Comment by lvlrSajjad** [comment]
> @lvlrSajjad we're removing Semaphore from this project and adding Circle CI instead. We can ignore the Semaphore failure and see whether it passes on Circle once it's ready.

**Comment by lvlrSajjad** [comment]
> @lvlrSajjad we're removing Semaphore from this project and adding Circle CI instead. We can ignore the Semaphore failure and see whether it passes on Circle once it's ready.

i just accidently closed the pull 
@mlaco i must fetch your circle config to being testable with circle ?

**Comment by mlaco** [comment]
@lvlrSajjad yes, you'll need the circle config, but it looks like you got it already.
We need to trigger Circle to test your branch. Please make an empty commit using
`git commit --allow-empty -m "fix(ci): Testing CI release"` to do so.

**Comment by lvlrSajjad** [comment]
> @lvlrSajjad yes, you'll need the circle config, but it looks like you got it already.
> We need to trigger Circle to test your branch. Please make an empty commit using
> `git commit --allow-empty -m "fix(ci): Testing CI release"` to do so.

Well i've done the commit even multipletimes but i can't see triggerness

**Comment by mlaco** [comment]
@lvlrSajjad you might need to create a new PR. Keep this one open, but can you also open a new one? I think it might trigger the build.

**Comment by lvlrSajjad** [comment]
Opened again

**Comment by mlaco** [comment]
@lvlrSajjad I found the setting that will allow Circle to run the tests on PRs

File: .circleci/config.yml [hunk 1]
```diff
@@ -50,22 +50,21 @@ jobs:
...
+    - restore_cache:
+        name: Restore node modules
+        keys:
+          - v1-dependencies-{{ checksum "package.json" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+    # Run semantic-release after all the above is set.
+    - run:
+        name: Publish to NPM
+        command: yarn ci:publish # this will be added to your package.json scripts
```
Comment (mlaco) [inline .circleci/config.yml]: @lvlrSajjad I think this is the problem. Looks like you merged in master before I fixed the indentation here. Please merge master again.

Comment (lvlrSajjad) [reply .circleci/config.yml]: i'm a bit noob to git things . can you tell me how to do this ?

Comment (mlaco) [reply .circleci/config.yml]: I think you got it; `config.yml` doesn't show up in the `files` anymore, which means your file matches ours.

**Comment by lvlrSajjad** [comment]
well it seems it can't find lint command in package.json of installed project . maybe it can't run ignite add standard

**Comment by mlaco** [comment]
> well it seems it can't find lint command in package.json of installed project . maybe it can't run ignite add standard

Yep. I'm currently trying to figure out why. It's able to do it on the master branch, so I'm kind of stumped.

**Comment by mlaco** [comment]
Congrats, it worked!

**Comment by lvlrSajjad** [comment]
@mlaco Test passed but i need to find the problem with patch in file :-D cause i need to do that modifications on mainactivity.java as this link says
https://reactnavigation.org/docs/en/getting-started.html

File: boilerplate.js [hunk 2]
```diff
@@ -175,11 +175,33 @@ async function install (context) {
...
+    await ignite.addModule('react-native-gesture-handler', { version: '1.0.9', link: true })
+    console.log(`${process.cwd()}/android/app/src/main/java/com/${name}/MainActivity.java`)
+    ignite.patchInFile(`${process.cwd()}/android/app/src/main/java/com/${name.toLowerCase()}/MainActivity.java`, {
+      after: 'import com.facebook.react.ReactActivity;',
+      insert: `
+      import com.facebook.react.ReactActivityDelegate;
+      import com.facebook.react.ReactRootView;
+      import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;`
+    })
+    //
```
Comment (markrickert) [inline boilerplate.js]: Should this commented out section be left in without an explanation of why it's commented out in favor of the code block above it?

File: boilerplate.js [hunk 3]
```diff
@@ -197,7 +219,7 @@ async function install (context) {
     }
 
     if (answers['redux-persist'] === 'Yes') {
-      await system.spawn(`ignite add redux-persist@"~>1.0.1" ${debugFlag}`, {
+      await system.spawn(`ignite add redux-persist@"~>5.10.0" ${debugFlag}`, {
```
Comment (markrickert) [inline boilerplate.js]: HOLY CRAP that's a massive version upgrade!

File: boilerplate.js (hunk 2 above)
Comment (lvlrSajjad) [reply boilerplate.js]: for now i'm just commenting / uncommenting stuff till i findout what is wrong i'll say when i'm done

**Comment by lvlrSajjad** [comment]
Guys I'm done 
@mlaco thank's for being patient and helping me to figure this out 
@markrickert thank's for reviews

**Comment by mlaco** [comment]
@lvlrSajjad yes, thank you! I appreciate your hard work and dedication.

**Pull request merged**